include/turtle/detail/expectation_template.hpp

```diff
@@ -196,16 +221,41 @@ namespace detail
         {
             return invocation_->verify();
         }
-
+#if defined(MOCK_THREAD_SAFE)
+        bool verify(const boost::shared_ptr< condition_variable> &cv, lock &lk) const
+        {
+            if (timeout_)
+            {
+                while (!invocation_->verify())
```

Comment:
So this can only be used with one expectation because it will block on that expectation ? What if test mock object has two methods with two expectations ?